### PR TITLE
Enhance GraphQL queries to includeSoftDelete parameter for relationships

### PR DESCRIPTION
**Pull request by murdo-moj**
* We're getting [errors](https://mojdt.slack.com/archives/C07SFN0MHRP/p1736258081094319) to do with the caseman database, as table descriptions are empty. There's a related issue which is that soft deleted tables are being listed in the database's relationships, which are removed by stateful ingestion
* This fix sets an empty string as a description for all tables in a database where there is no description
* The parameter to exclude soft deleted relationships is also there, but it doesn't seem to work

### CODE REVIEW

File: lib/datahub-client/data_platform_catalogue/client/parsers.py [hunk 1]
```diff
@@ -430,9 +430,12 @@ def parse_relations(
                     if relation.get("entity", {}).get("properties") is not None
                     else relation.get("entity").get("name", "")
                 )
+                properties = relation.get("entity", {}).get("properties")
+                description_field = relation.get("entity").get("properties", {}).get("description", "")
```
Comment (MatMoore) [inline lib/datahub-client/data_platform_catalogue/client/parsers.py:434]: It looks like this code fails on some cases that the previous one doesn't, for example `{"entity": {"properties": None}}` triggers `AttributeError: 'NoneType' object has no attribute 'get'`

Elsewhere in this module we have expressions like `entity.get("properties") or {}` to handle both missing and None values. There might be a cleaner way than this but it should remove the edge cases.

If it helps, I wrote a hypothesis test (with some help from ChatGPT):

```python
from hypothesis import given
from hypothesis import strategies as st

from parser_change import after, before

strategy = st.fixed_dictionaries(
    {
        "entity": st.one_of(
            st.just({}),
            st.fixed_dictionaries(
                {
                    "properties": st.one_of(
                        st.none(),
                        st.dictionaries(
                            keys=st.sampled_from(["description", "banana"]),
                            values=st.one_of(st.none(), st.text()),
                        ),
                    )
                }
            ),
        )
    }
)


@given(strategy)
def test_equivalent(s):
    try:
        previous_value = before(s)
    except Exception:
        pass
    else:
        if previous_value is None:
            assert after(s) == ""
        else:
            # New version should be equivalent to old version
            assert previous_value == after(s)
```

Comment (murdo-moj) [reply lib/datahub-client/data_platform_catalogue/client/parsers.py]: I've changed the implementation to account for properties being filled as none, and description being filled as none. In both cases we want a value of "" for description

File: lib/datahub-client/data_platform_catalogue/client/parsers.py [hunk 2]
```diff
@@ -430,11 +428,11 @@ def parse_relations(
...
                     else relation.get("entity").get("name", "")
                 )
-                description = (
-                    relation.get("entity").get("properties", {}).get("description", "")
-                    if relation.get("entity", {}).get("properties") is not None
-                    else ""
-                )
+                properties = relation.get("entity", {}).get("properties")
+                if properties is not None:
+                    description = properties.get("description") or ""
```
Comment (mitchdawson1982) [inline lib/datahub-client/data_platform_catalogue/client/parsers.py:433]:
```suggestion
                    description = properties.get("description", "")
```

File: lib/datahub-client/data_platform_catalogue/client/parsers.py [hunk 3]
```diff
@@ -430,11 +428,11 @@ def parse_relations(
...
+                elif properties is None:
+                    description = ""
```
Comment (mitchdawson1982) [inline lib/datahub-client/data_platform_catalogue/client/parsers.py:435]: We have tested for not None in the `if`
```suggestion
                else:
                    description = ""
```

File: lib/datahub-client/data_platform_catalogue/client/parsers.py (hunk 2 above)
Comment (murdo-moj) [reply lib/datahub-client/data_platform_catalogue/client/parsers.py:433]: sometimes we have {"description": None}, which we want to resolve to `""`, and would with this change resolve to `None`

**Review by mitchdawson1982 — APPROVED** [review]
